iib/workers/tasks/utils.py

```diff
@@ -19,14 +21,83 @@
 )
 from operator_manifest.operator import ImageName
 
-from iib.exceptions import IIBError
+from iib.exceptions import IIBError, AddressAlreadyInUse
 from iib.workers.config import get_worker_config
 from iib.workers.api_utils import set_request_state
 
 log = logging.getLogger(__name__)
 dogpile_cache_region = create_dogpile_region()
 
 
+def _add_property_to_index(db_path, property):
+    """
+    Add a property to the index.
+
+    :param str db_path: path to the index database
+    :param dict property: a dict representing a property to be added to the index.db
+    """
+    insert = (
+        'INSERT INTO properties '
+        '(type, value, operatorbundle_name, operatorbundle_version, operatorbundle_path) '
+        'VALUES (?, ?, ?, ?, ?);'
+    )
+    con = sqlite3.connect(db_path)
+    # Insert property
+    con.execute(
+        insert,
+        (
+            property['type'],
+            property['value'],
+            property['operatorbundle_name'],
+            property['operatorbundle_version'],
+            property['operatorbundle_path'],
+        ),
+    )
+    con.commit()
+    con.close()
+
+
+def add_max_ocp_version_property(resolved_bundles, temp_dir):
+    """
+    Add the max ocp version property to bundles.
+
+    We need to ensure that any bundle which has deprecated/removed API(s) in 1.22/ocp 4.9
+    will have this property to prevent users from upgrading clusters to 4.9 before upgrading
+    the operator installed to a version that is compatible with 4.9
+
+    :param list resolved_bundles: list of resolved bundles to which the max ocp version property
+        will be added if missing
+    :param str temp_dir: directory location of the index image
+    """
+    # Get the CSV name and version (not just the bundle path)
+    db_path = temp_dir + "/database/index.db"
+    port, rpc_proc = serve_index_registry(db_path)
+
+    raw_bundles = run_cmd(
+        ['grpcurl', '-plaintext', f'localhost:{port}', 'api.Registry/ListBundles'],
+        exc_msg='Failed to get bundle data from index image',
+    )
+    rpc_proc.kill()
+
+    # Get bundle json for bundles in the request
+    updated_bundles = list(
```

Comment:
Could you update a comment here?
`Get a bundlePath from raw_bundles and check the presence in resolved_bundles to return a list of present bundles`
I am lost here. You convert `raw_bundles` to JSON and then you check presence with resolved_bundles and create a LIST of `updated_bundles` but then you try to work with the `updated_bundles` as a DICTIONARY.

Reply:
I check filter() here: https://www.programiz.com/python-programming/methods/built-in/filter

Reply:
I have been testing it.